### PR TITLE
Bug 1932281:  inform user when channels are unknown

### PR DESCRIPTION
**Pull request by rhamilto**
if `!clusterVersion.spec.channel`, add alert instructing to configure a channel and improve Channel empty state indicator.
Screen Shot 2021-03-29 at 3 26 24 PM

if `clusterVersion.status.desired.channels`
Screen Shot 2021-03-29 at 3 24 20 PM

if `!clusterVersion.status.desired.channels`
Screen Shot 2021-03-29 at 3 23 25 PM
Screen Shot 2021-03-29 at 3 23 39 PM

if version is not found in `clusterVersion.spec.channel`
Screen Shot 2021-03-29 at 3 26 14 PM

### CODE REVIEW

**Review by zherman0 — APPROVED** [review]

File: frontend/public/components/modals/cluster-channel-modal.tsx [hunk 1]
```diff
@@ -22,34 +29,67 @@ const ClusterChannelModal = withHandlePromise((props: ClusterChannelModalProps)
     o[val] = val;
     return o;
   }, {});
+  const version = semver.parse(getLastCompletedUpdate(cv));
   const submit: React.FormEventHandler<HTMLFormElement> = (e) => {
     e.preventDefault();
     const patch = [{ op: 'add', path: '/spec/channel', value: channel }];
     return handlePromise(k8sPatch(ClusterVersionModel, cv, patch), close);
   };
+  const channelsExist = cv.status.desired.channels;
```
Comment (spadgett) [inline frontend/public/components/modals/cluster-channel-modal.tsx:38]: I'd be careful and use optional chaining.

```suggestion
  const channelsExist = cv.status?.desired?.channels?.length;
```

**Review by spadgett — COMMENTED** [review]
LGTM, just one nit to avoid a runtime error if the `status` stanza isn't yet defined. I'm not sure if this would ever happen in a real cluster, but I would code defensively just in case since it's not a required field in the API definition.

File: frontend/public/components/modals/cluster-channel-modal.tsx (hunk 1 above)
Comment (rhamilto) [reply frontend/public/components/modals/cluster-channel-modal.tsx]: Thanks, @spadgett.  Updated.

**Comment by rhamilto** [comment]
Added an integration test fix.

**Review by spadgett — APPROVED** [review]
/lgtm

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**